File: core/src/com/inkostilation/pong/commands/response/ResponseGameStateCommand.java
package com.inkostilation.pong.commands.response;

import com.inkostilation.pong.commands.AbstractResponseCommand;
import com.inkostilation.pong.engine.GameState;

public class ResponseGameStateCommand extends AbstractResponseCommand {

    private GameState state;

    public ResponseGameStateCommand(GameState state) {
        this.state = state;
    }

    @Override
    public void execute() {
        getNotifier().notifyObservers(state);
    }
}

File: core/src/com/inkostilation/pong/commands/response/ResponseGameStateCommand.kt
package com.inkostilation.pong.commands.response

import com.inkostilation.pong.commands.AbstractResponseCommand
import com.inkostilation.pong.engine.GameState

class ResponseGameStateCommand(private val state: GameState) : AbstractResponseCommand() {
    override fun execute() {
        notifier.notifyObservers(state)
    }

}